### DISCUSSION TITLE
Creating a GUI for prolif

### DISCUSSION THREAD

**Discussion by pratik-narain** [chemosim-lab/ProLIF#36] [Ideas]
I am willing to create a PyQT based gui for prolif as it fills the niche of LigPlot very easily and can be a good competitor. It is even better as it works on MD trajectories. The only thing I know of which performs the same function if the SID module of maestro but its not free. I have done a mockup based on the LigPlot gui. Would you like to participate? Currently I am trying to figure out a way to package the whole thing as a binary app for linux and windows (don't have access to a mac). But it seems like packaging a whole conda environment is not possible as of now. I am a super beginner at app development but do have coding experience for scientific problems.

**Comment by cbouy** [comment]
Hi @pratik-narain,

Thanks for your interest in developing a GUI for ProLIF.
While having a LigPlot equivalent based on prolif sounds great, I just want to make sure you realize that depending on what you want the app to perform, it's going to be a pretty big commitment.
It's unlikely that I'll have free time to contribute to the coding and testing part, but I should be able to give you some tips and review the code as I've developed a PyQt app in the past.
You might want to have a look at [PyInstaller](https://pyinstaller.readthedocs.io/en/stable/) for packaging the PyQt app as a single executable, it worked pretty well for me before.

So a few questions for you:
- What do you want the app to do? Read an MD traj and output a LigPlot-like diagram? Read any input? Output over kinds of info?
- Do you have any experience with writing unit-tests (preferably with pytest)?

Have a great day,
Cédric

**Reply by pratik-narain** [reply]
Indeed, I would like to keep it simple at first. Just read a PDB (with ligand) or a topology/trajectory pair and output a ligand interaction diagram like LigPlot. Once this is achieved, it can be expanded to other functionalities similar to the SID module of schrodinger an output a well written PDF/HTML report.

https://www.schrodinger.com/training/videos/molecular-dynamics/simulation-interactions-diagram-generate-report
